### ISSUE TITLE
Export functions for images and uploads don't work, nothing downloads.

### ISSUE THREAD

**Issue by Pedrotheplant** [nikobojs/manuscrape_electron#72]
Doesn't work in neither Firefox nor Edge

**Comment by nikobojs** [comment]
It works perfectly here. Can you test on another project? Maybe the bug is related directly to the data in your project. Thanks !

**Comment by Pedrotheplant** [comment]
Works on other projects, maybe it is the size?

**Comment by nikobojs** [comment]
I seems to work in general, but for bigger projects the requests take a very long time (if too long, it will return an error).

The server was also running out of ram and utilizing the swap partition. That makes the server very slow. I upgraded the server hardware and gave it a restart. So the bug should be kind of fixed already.

A possible permanent fix is to compress the images to some maximum width when they're getting uploaded. How does that sound?

**Comment by Pedrotheplant** [comment]
Sounds good - lets maybe run a test, need to check that the images a legible when downloaded again

**Comment by nikobojs** [comment]
I made the downloading from the fileserver work in parallel while streaming directly into a zip file, which is streamed directly to the client. Will do some benchmarking soon and close the issue if the performance is smooth

**Comment by nikobojs** [comment]
So I got a large download from 30 seconds down to 12 seconds. I also added some UI that blocks the user from downloading/exporting multiple times in parallel. Hopefully this fixes the problem. If not, please create another issue.
